engine/include/Engine/Core/Rendering/Window/IWindow.hpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include <string>
+
+namespace Engine::Core::Renderering
+{
+struct WindowCreateArg
+{
+    std::string name;
+    size_t height;
+    size_t width;
+};
+
+class IWindow
```

Comment:
I don't think an abstraction of the window is necessary. I don't see us letting the engine user choosing whether they want to use the SDL, GLFW, X11, etc.. Since the way we manage windows is "internal" to the engine and not meant to be exposed to the developer, I wouldn't make it too intricate.

Much like Unreal chose SDL2 or like Unity chose to make their own backend, we picked GLFW. I believe we should build around it directly.

Pinging you @ThomasProg, your insight on this would be appreciated.

Reply:
If we want to use a specific window we can directly use an GLFWWindow. I don't think that its will be a bottleneck. I am also for the idea to wrap lib in an interface to easily change it if the implementation of the lib turn out to be too restrictive

Reply:
I think that being able to change between SDL, GLFW etc should be done. 
It was also asked in the project.
However, I think static polymorphism is sufficient since we only have to change the library we're using at compile time.
In short, for example : 
- We can do a class GLFWWindow, having the same functions than SDLWindow
- We can put "using Window = GLFWWindow;" or "using Window = SDLWindow;" depending on a macro
Then, we could create our window the same way every time without problem.
A runtime polymorphism is also good for me, but at the end we want to use one and only one window library at the same time, so we should define subclasses only when a library is imported.
However, that would mean we don't need runtime polymorphism.